hack/manifest-gen/main.go

```diff
@@ -0,0 +1,145 @@
+	cmd := buildCmd()
+	if err := cmd.Execute(); err != nil {
+		log.Printf("Error: %v", err)
+		os.Exit(1)
+	}
+}
+
+func buildCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:           "manifest-gen",
+		Short:         "ECK manifest generator",
+		Long:          `Utilities for generating an installation manifest for ECK`,
+		SilenceErrors: true,
+		SilenceUsage:  true,
+	}
+
+	sourceDir, err := findSourceDir()
+	if err != nil {
+		log.Fatalf("Failed to locate assets: %v", err)
+	}
+
+	cmd.PersistentFlags().StringVar(&sourceFlag, "source", sourceDir, "Source directory")
+	_ = cmd.PersistentFlags().MarkHidden("source")
+
+	cmd.AddCommand(generateCmd())
+	cmd.AddCommand(optionsCmd())
+
+	return cmd
+}
+
+func findSourceDir() (string, error) {
+	execPath, err := os.Executable()
+	if err != nil {
+		return "", err
+	}
+
+	path, err := filepath.Abs(execPath)
+	if err != nil {
+		return "", err
+	}
+
+	return filepath.Join(filepath.Dir(path), "assets", "charts", "eck"), nil
+}
+
+func generateCmd() *cobra.Command {
+	desc := `
+Generates a manifest for installing ECK.
+
+There are two pre-defined profiles for installing ECK. The "global" profile installs ECK with access to the
+whole Kubernetes cluster. The "restricted" profile installs ECK restricted to a single (or several) namespace(s).
+
+The generated manifest can be customized using the "--set" and "--values" flags. Use the "options" command to list all
+available configuration options. The behaviour of these flags is identical to the similarly named flags used by Helm.
+See https://helm.sh/docs/intro/using_helm/ for more information.
+
+By default, the operator is installed into the "elastic-system" namespace. This can be overridden by setting the
+"operator.namespace" option.
+
+`
+	examples := `
+Global operator:
+    $ manifest-gen generate
+
+Global operator with the validation webhook disabled:
+    $ manifest-gen generate --set=config.webhook.enabled=false
+
+Global operator with resource memory limit increased to 300Mi and CPU limit increased to 2:
+    $ manifest-gen generate --set=operator.resources.limits.cpu=2,operator.resources.limits.memory=300Mi
+
+Restricted operator without CRDs, managing the "elastic-system" namespace:
+    $ manifest-gen generate --profile=restricted --exclude-crds
+
+Restricted operator installed to and managing the single namespace named "namespacex":
+    $ manifest-gen generate --profile=restricted --set=operator.namespace=namespacex --set=config.managedNamespaces='{namespacex}'
+
+Restricted operator managing "elastic-system", "nsa" and "nsb":
+    $ manifest-gen generate --profile=restricted --set=config.managedNamespaces='{elastic-system, nsa, nsb}'
+
+Restricted operator with tracing configured:
+    $ manifest-gen generate --profile=restricted --set=config.tracing.enabled=true --set=config.tracing.config.ELASTIC_APM_SERVER_URL=http://apm:8200
+`
+
+	cmd := &cobra.Command{
+		Use:           "generate",
+		Short:         "Generate ECK manifests",
```

Comment:
We're a bit inconsistent in using `manifests` vs. `manifest`. Not sure what's best? Let's try to settle on one.

Reply:
I guess it's the awkwardness of using the plural form for names. `manifests-gen` doesn't sound right -- hence why sometimes `manifest` is used in places.